sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateStoreIntegrationSuite.scala

```diff
@@ -223,6 +223,48 @@ class RocksDBStateStoreIntegrationSuite extends StreamTest
     }
   }
 
+  testWithColumnFamilies("SPARK-51823: unload state stores on commit",
+    TestWithBothChangelogCheckpointingEnabledAndDisabled) { colFamiliesEnabled =>
+    withTempDir { dir =>
+      withSQLConf(
+        (SQLConf.STATE_STORE_PROVIDER_CLASS.key -> classOf[RocksDBStateStoreProvider].getName),
+        (SQLConf.CHECKPOINT_LOCATION.key -> dir.getCanonicalPath),
+        (SQLConf.SHUFFLE_PARTITIONS.key -> "1"),
+        (SQLConf.STATE_STORE_UNLOAD_ON_COMMIT.key -> "true")) {
+        val inputData = MemoryStream[Int]
+
+        val query = inputData.toDS().toDF("value")
+          .select($"value")
+          .groupBy($"value")
+          .agg(count("*"))
+          .writeStream
+          .format("console")
+          .outputMode("complete")
+          .start()
+        try {
+          inputData.addData(1, 2)
+          inputData.addData(2, 3)
+          query.processAllAvailable()
+
+          // StateStore should be unloaded, so its tmp dir shouldn't exist
+          for (file <- new File(Utils.getLocalDir(sparkConf)).listFiles()) {
+            assert(!file.getName().startsWith("StateStore"))
```

Comment:
Oof this only works if I run this test by itself. When running the whole Suite, all the other tests leave StateStore data on disk so this check sees those

Reply:
Fixed by adding an `afterEach` to clear the state store. The parent `StreamTest` only does an `afterAll` to clear the state store, not sure if that one should just be updated to `afterEach` instead

Reply:
Still doesn't work, too many other suites not cleaning up after themselves, need to think about how else to verify